src/third-party/PhishingController.ts

```diff
@@ -133,30 +136,44 @@ export class PhishingController extends BaseController<
     };
     this.detector = new PhishingDetector(this.defaultState.phishing);
     this.initialize();
-    this.poll();
   }
 
   /**
-   * Starts a new polling interval.
+   * Set the interval at which the phishing list will be refetched. Fetching will only occur on the next call to test/bypass. For immediate update to the phishing list, call updatePhishingLists directly.
+   * last this.lastFetched.
    *
-   * @param interval - Polling interval used to fetch new approval lists.
+   * @param interval - the new interval, in ms
    */
-  async poll(interval?: number): Promise<void> {
-    interval && this.configure({ interval }, false, false);
-    this.handle && clearTimeout(this.handle);
-    await safelyExecute(() => this.updatePhishingLists());
-    this.handle = setTimeout(() => {
-      this.poll(this.config.interval);
-    }, this.config.interval);
+  setRefreshInterval(interval: number) {
+    this.configure({ refreshInterval: interval }, false, false);
+  }
+
+  /**
+   * Calls this.updatePhishingLists if this.refreshInterval has passed since last this.lastFetched.
+   *
+   * @returns Promise<void> when finished fetching phishing lists or when fetching in not necessary.
+   */
+  private async fetchIfNecessary(): Promise<boolean> {
+    const outOfDate =
+      Date.now() - this.lastFetched >= this.config.refreshInterval;
+
+    if (this.lastFetched === 0 || outOfDate) {
+      await safelyExecute(() => this.updatePhishingLists());
+      this.lastFetched = Date.now();
+      return true;
```

Comment:
It doesn't seem like the return value of this method is ever used. Maybe it is unnecessary to return anything?